util/test/computePerfStats

```diff
@@ -1,232 +1,312 @@
+            if header == None: # for all lines after the header
+                # ignore comments and empty lines
+                if parsed[0] == "" or "#" in parsed[0]:
+                    continue
+                if len(parsed) != columns:
+                    print("[Error: {} entry for {} doesn't match header length.]"
+                            .format(data_file, parsed[0]))
+                    exit(2)
+
+        if header == None: # not found
+            print("[Error: {} has no header.]".format(data_file))
+            exit(2)
+
+    
+    keys_actual = [k.strip() for k in keys_actual]
+    keys_exp = [k.strip() for k in keys_exp]
+
+    s = difflib.SequenceMatcher(None, keys_actual, keys_exp)
+    diff = s.get_opcodes()
+
+    # the two aren't equal
+    if len(diff) > 1 or (len(diff) == 1 and diff[0][0] != "equal"):        
+        print("[Error: key mismatch between {} and {}]"
+                .format(args.keys_file, data_file))
+        print("Changes since creation of .dat file:")
+        for d in diff:
+            type = d[0]
+            if type == "equal":
+                continue
+            elif type == "delete":
+                for i in range(d[1], d[2]):
+                    print("\t'{}' has been removed.".format(keys_actual[i]))
+            elif type == "insert":
+                for i in range(d[3], d[4]):
+                    print("\t'{}' has been added.".format(keys_exp[i]))
+            elif type == "replace":
+                if d[2] - d[1] == 1: # singlular
+                    print("\t'{}' has been replaced with '{}'.".format(
+                            keys_actual[d[1]], ", ".join(keys_exp[d[3]:d[4]])))
+                else: # plural
+                    print("\t'{}' have been replaced with '{}'.".format(
+                        ", ".join(keys_actual[d[1]:d[2]]), 
+                        ", ".join(keys_exp[d[3]:d[4]])))
+
+        exit(2)
+
+
+def setup(args):
+    # finish setting up command line args
+    if not args.keys_file:
+        args.keys_file = "%s.perfkeys" % args.testname;
+    if not args.test_output_file:
+        args.test_output_file = "%s.exec.out.tmp" % args.testname;
+    if not args.perfdate:
+        args.perfdate = time.strftime("%m/%d/%y");
+        print("Using default date %s" % args.perfdate);
+    else:
+        print("Using set date %s" % args.perfdate);
+
+    if args.exec_time_out == "True":
+        args.exec_time_out == True
+    else:
+        args.exec_time_out == False
+    
+    global data_file, errors_file, fatal_errors;
+    data_file = "%s/%s.dat" % (args.outputdir, args.testname)
+    errors_file = "%s/%s.errors" % (args.outputdir, args.testname)
+    fatal_errors = 0;
+
+    if os.path.isfile(errors_file):
+        os.remove(errors_file)
+    
+
+def cleanup():
+    print("output was: ")
+    # if test output was too long, only print first and last 1000 characters
+    if len(test_output_raw) > 2000: 
+        print(test_output_raw[:1000].strip())
```

Comment:
Might have to explain this comment a little better, I'm not sure if it'll be obvious what you're talking about to Cory.